src/components/opt-in-out/index.tsx

```diff
@@ -1,71 +1,48 @@
-import {
-  PlatformOptionRedirectData,
-  OptInOutProps,
-  isOptInPlatformOption,
-} from './types'
 import { safeAnalyticsTrack } from 'lib/analytics'
-import makeBetaWelcomeToast from './helpers/make-beta-welcome-toast'
-import { getIoRedirectPath } from './helpers/get-io-redirect-path'
+import { OptInOutProps, isOptInPlatformOption } from './types'
+import { PLATFORM_OPTIONS, postFormData, makeBetaWelcomeToast } from './helpers'
+import { OptOutFormState } from './components/opt-out-form/types'
+import { OptOutButtonAndDialog } from './components/opt-out-button-and-dialog'
 
-// Could these go in the config? or I could source the base urls elsewhere
-export const PLATFORM_OPTIONS: PlatformOptionRedirectData = {
-  learn: {
-    base_url: 'https://learn-git-ksspike-opt-in-redirects-hashicorp.vercel.app', // FOR TESTING PURPOSES NEED TO UPDATE for - 'https://learn.hashicorp.com/'
-    getRedirectPath: getLearnRedirectPath,
-    cookieKey: 'learn-beta-opt-in',
-    cookieAnalyticsKey: 'learn-beta-opt-in-tracked',
-  },
-  'waypoint-io': {
-    base_url: 'https://www.waypointproject.io/',
-    getRedirectPath(path) {
-      return this.base_url + getIoRedirectPath(path)
-    },
-    cookieKey: 'waypoint-io-beta-opt-in',
-    cookieAnalyticsKey: 'waypoint-io-beta-opt-in-tracked',
-  },
-  'vault-io': {
-    base_url: 'https://www.vaultproject.io/',
-    getRedirectPath(path) {
-      return this.base_url + getIoRedirectPath(path)
-    },
-    cookieKey: 'vault-io-beta-opt-in',
-    cookieAnalyticsKey: 'vault-io-beta-opt-in-tracked',
-  },
-}
-
-export default function OptInOut({ platform, redirectPath }: OptInOutProps) {
+export default function OptInOutController({
+  platform,
+  redirectPath,
+}: OptInOutProps) {
   // fire toast, render button, etc
   const router = useRouter()
-  const [showDialog, setShowDialog] = useState(false)
   const optedIn = Cookies.get(PLATFORM_OPTIONS[platform].cookieKey)
-
-  const openDialog = () => setShowDialog(true)
-  const closeDialog = () => setShowDialog(false)
   const url =
     redirectPath || PLATFORM_OPTIONS[platform].getRedirectPath(router.asPath)
 
   /**
    * Handle opt out, which is passed to our opt out form,
    * and is also used in our welcome toast.
    */
-  const handleOptOut = useCallback(() => {
-    // @TODO - handle form submit
-    Cookies.remove(PLATFORM_OPTIONS[platform].cookieKey)
-    Cookies.remove(PLATFORM_OPTIONS[platform].cookieAnalyticsKey)
-    safeAnalyticsTrack('Beta Opted Out', {
-      bucket: platform,
-    })
-    window.location.assign(url)
-  }, [platform, url])
+  const handleOptOut = useCallback(
+    async (state: OptOutFormState) => {
+      await postFormData({
+        segment_anonymous_id: uuid(),
+        primary_opt_out_reason: state.optOutReason,
+        details: state.optOutDetails,
+        opt_out_page_url: new URL(
+          router.asPath,
+          __config.dev_dot.canonical_base_url
+        ).toString(),
+      })
```

Comment:
we should wrap this in a `try/catch`, maybe log if call fails? or just no-op, but then we won't get any unexpected behavior.